lib/extractor/default.nix

```diff
@@ -114,7 +114,7 @@ let
     chmod +x setup.py || true
     mkdir $out
     echo "extracting dependencies"
-    out_file=$out/python.json ${py}/bin/python -c "${setuptools_shim}" install &> $out/python.log || true
+    SETUPTOOLS_USE_DISTUTILS=stdlib out_file=$out/python.json ${py}/bin/python -c "${setuptools_shim}" install &> $out/python.log || true
```

Comment:
I was sure I had tested this [already](https://github.com/DavHau/mach-nix/issues/467#issuecomment-1156525579)(!).

I'll have to test again.

In the meantime, perhaps add something like as a comment above the changed line?

```
# Restore old setuptools behaviour (since
# https://github.com/pypa/setuptools/commit/b6fcbbd00cb6d5607c9272dec452a50457bdb292),
# to keep it working with mach-nix. 
```

Reply:
> I'll have to test again.

It still fails for me :-(

Reply:
I must admit, I placed your derivation in my flake for testing and it was working from my local folder,
but it also fails for me from the github pr. Not sure what's going on yet.

Reply:
> edit: yes, there's another regression down the line. back to bisecting...

Right, it's coming back to me now! (But don't remember what _that_ was about -- I didn't bisect it.)

Reply:
bisected it to nixpkgs 7f802c704651baba3adb2abdd9cc6645c987b9ad, which changed 'something' about the python wrapper (made it a binary for darwin?). Diving in.

Reply:
yeah, of course,
```
# fix executables
        for f in $(ls ${python_env}/bin); do
          sed -i "s|${python_env}|$out|g" $out/bin/$f
          sed -i "/NIX_PYTHONPATH/a export PYTHONPATH=$out\/lib\/python${major}.${minor}" $out/bin/$f
        done
```
is going to mess up that binary wrapper.

Reply:
Doh!

Reply:
The good: it works. The bad: it's ugly! :-)

Reply:
Do we really need to extract the wrapper info, or can we just "start from scratch"?

It seems the original code hardcoded a PYTHONPATH= value, but left the rest alone. Doesn't that mean we can wrap the wrapper and in our outer wrapper set PYTHONPATH= and leave the other variables to be set by the inner wrapper?

Reply:
If the double wrapper works, I think we can use it for the non-binary wrapper too -- just one code path instead of two. 

This double wrapper is only used for extracting info from the package, right? If so the "ugliness" should be contained. (It'd be worse IMHO if the python envs exposed to users had to be double wrapped.)

Reply:
All right, the CI says this works, so once more with feeling.

We now only have as single code path, and we fall back to the non-binary wrapper if makeBinaryWrapper is missing from nixpkgs (pre nixpkgs 22.05).

And I agree, it's limited to the dependency extractor in the first place.

This is a much cleaner solution than the advanced string munging.

Reply:
Awesome!